sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -111,6 +113,52 @@ object RapidsPluginUtils extends Logging {
     }
   }
 
+  private def detectMultipleJar(propName: String, jarName: String, conf: RapidsConf): Unit = {
+    val classloader = ShimLoader.getShimClassLoader()
+    val possibleRapidsJarURLs = classloader.getResources(propName).asScala.toSet.toSeq.filterNot {
+      url => url.toString.contains("test")
+    }
+    val revisionMap: Map[String, Seq[URL]] = possibleRapidsJarURLs.map { url =>
+      val versionInfo = scala.io.Source.fromURL(url).getLines().toSeq
+      lazy val revision = versionInfo
+        .filter(_.startsWith("revision="))
+        .map(_.split("=").last)
```

Comment:
if you define a regex
```scala
val revisionRegex = "revision=(.*)".r
```

you could do it in one shot
```suggestion
        .collect { 
          case revisionRegex(revision) => revision 
        }
```